OpenSSL before 3.2 doesn't support deterministic ECDSA
and the test suite does not account for that.

Partially backported from usptream:
https://github.com/rthalley/dnspython/commit/fa92e09c55c79b625a66b8a242d6d71e02f21e99

--- dnspython-2.8.0/tests/test_dnssec.py
+++ dnspython-2.8.0/tests/test_dnssec.py
@@ -44,8 +44,17 @@
 except ImportError:
     pass  # Cryptography ImportError already handled in dns.dnssec
 
+    def default_backend():
+        raise NotImplementedError
+
+
 # pylint: disable=line-too-long
 
+if dns.dnssec._have_pyca:
+    have_deterministic = default_backend().ecdsa_deterministic_supported()
+else:
+    have_deterministic = False
+
 abs_dnspython_org = dns.name.from_text("dnspython.org")
 
 abs_keys = {
@@ -1412,10 +1421,12 @@ def testSignatureRSASHA256(self):  # type: () -> None
         )
         self._test_signature(key, dns.dnssec.Algorithm.RSASHA256, abs_soa)
 
+    @unittest.skipUnless(have_deterministic, "deterministic ECDSA not available")
     def testSignatureECDSAP256SHA256(self):  # type: () -> None
         key = ec.generate_private_key(curve=ec.SECP256R1(), backend=default_backend())
         self._test_signature(key, dns.dnssec.Algorithm.ECDSAP256SHA256, abs_soa)
 
+    @unittest.skipUnless(have_deterministic, "deterministic ECDSA not available")
     def testDeterministicSignatureECDSAP256SHA256(self):  # type: () -> None
         key = ec.generate_private_key(curve=ec.SECP256R1(), backend=default_backend())
         inception = time.time()
@@ -1454,6 +1465,7 @@ def testNonDeterministicSignatureECDSAP256SHA256(self):  # type: () -> None
         )
         assert rrsigset1 != rrsigset2
 
+    @unittest.skipUnless(have_deterministic, "deterministic ECDSA not available")
     def testSignatureECDSAP384SHA384(self):  # type: () -> None
         key = ec.generate_private_key(curve=ec.SECP384R1(), backend=default_backend())
         self._test_signature(key, dns.dnssec.Algorithm.ECDSAP384SHA384, abs_soa)
--- dnspython-2.8.0/tests/test_dnssecalgs.py
+++ dnspython-2.8.0/tests/test_dnssecalgs.py
@@ -42,6 +42,20 @@
 except ImportError:
     pass  # Cryptography ImportError already handled in dns.dnssec
 
+try:
+    from cryptography.hazmat.backends import default_backend
+except ImportError:
+    pass  # Cryptography ImportError already handled in dns.dnssec
+
+    def default_backend():
+        raise NotImplementedError
+
+
+if dns.dnssec._have_pyca:
+    have_deterministic = default_backend().ecdsa_deterministic_supported()
+else:
+    have_deterministic = False
+
 
 @unittest.skipUnless(dns.dnssec._have_pyca, "Python Cryptography cannot be imported")
 class DNSSECAlgorithm(unittest.TestCase):
@@ -92,6 +106,7 @@ def test_dsa(self):
             k = PrivateDSA.generate(2048)
             k.sign(b"hello")
 
+    @unittest.skipUnless(have_deterministic, "deterministic ECDSA not available")
     def test_ecdsa(self):
         self._test_dnssec_alg(PrivateECDSAP256SHA256)
         self._test_dnssec_alg(PrivateECDSAP384SHA384)
